utils/src/main/kotlin/Utils.kt

```diff
@@ -282,7 +282,7 @@ fun Throwable.collectMessages(): List<String> {
     var cause: Throwable? = this
     while (cause != null) {
         messages += "${cause.javaClass.simpleName}: ${cause.message}"
-        cause = cause.cause
+        cause = cause.cause.takeIf { it != cause }
```

Comment:
This is not necessary as `getCause()` never returns `this`:

```java
public synchronized Throwable getCause() {
    return (cause==this ? null : cause);
}
```

If we implement this the reason should be to avoid an endless loop in case someone overrides `getCause()` to return `this`.

Reply:
Good point. And I was wondering why we weren't actually running into the endless loop issue 😆 I'll drop the commit.